juju/api.go

```diff
@@ -556,3 +429,85 @@ func addrsChanged(a, b []string) bool {
 	}
 	return false
 }
+
+var cachedAddressesExistErr = errors.New("cached API endpoints unexpectedly exist")
+
+// ControllerUpdateParams holds controller details for the UpdateControllerAddresses call.
+type ControllerUpdateParams struct {
+	ControllerName string
+	ControllerUUID string
+	CACert         string
+}
+
+// UpdateControllerAddresses writes any new api addresses to the client controller file.
```

Comment:
I'm -1 on having this function allow non-existing controllers, and having two ways to identify controllers. It should just do the address updates, and AFAIK there should be no need to identify by UUID.

I think register is the only place where the controller doesn't exist, right? Can we just create the controller details in register with no addresses set, and then call this?